core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1431,6 +1435,46 @@ protected void execute(UnifiedLog log) throws InterruptedException, RemoteStorag
             log.updateHighestOffsetInRemoteStorage(offsetAndEpoch.offset());
         }
     }
+    
+    private static boolean deleteRemoteLogSegment(
```

Comment:
Is there a reason for this method to be `static`? If non-static, then we don't need the below parameters:

1. remoteLogMetadataManager
2. remoteLogStorageManager
3. brokerTopicStats
4. time and
5. brokerId

Reply:
Originally, this method was only used by the expiration task. Now it is also used by the copy task in order to keep the segment state updated with the deletion attempt. We could create an helper class, but I thought it was too much for a single method. Wdyt?

Reply:
can we make the method as non-static and remove those parameters? They are field variables in the class.

Reply:
Oh yes, done.